src/gluonnlp/data/data_preprocessing_transform.py

```diff
@@ -0,0 +1,515 @@
+                 start_token=None,
+                 end_token=None):
+        self._vocab = vocab
+        self._start_token = start_token
+        self._end_token = end_token
+        self._token_after_seg = token_after_seg if token_after_seg else []
+
+    def __call__(self, seqs):
+        assert isinstance(seqs, collections.abc.Iterable) and len(seqs) > 0
+        tokens = []
+        if self._start_token:
+            tokens.append(self._start_token)
+        tokens_a = seqs if not isinstance(
+            seqs[0], collections.abc.Iterable) else seqs[0]
+        tokens.extend(tokens_a)
+        p_mask = [0] * len(tokens)
+        sp_token_counter = 0
+
+        if sp_token_counter < len(self._token_after_seg):
+            if self._token_after_seg[sp_token_counter]:
+                tokens.append(self._token_after_seg[sp_token_counter])
+            sp_token_counter += 1
+
+        segment_ids = [0] * len(tokens)
+        p_mask.append(1)
+        seqs = seqs[1:]
+
+        if len(seqs) > 0 and isinstance(seqs[0], collections.abc.Iterable):
+            # if seqs is a list of sequence
+            for (i, seq) in enumerate(seqs):
+                tokens_b = seq
+                tokens.extend(tokens_b)
+                p_mask.extend([0] * (len(tokens) - len(p_mask)))
+                if sp_token_counter < len(self._token_after_seg):
+                    if self._token_after_seg[sp_token_counter]:
+                        tokens.append(self._token_after_seg[sp_token_counter])
+                        p_mask.append(1)
+                    sp_token_counter += 1
+                segment_ids.extend([i + 1] * (len(tokens) - len(segment_ids)))
+
+        if self._end_token:
+            tokens.append(self._end_token)
+            p_mask.append(0)
+            segment_ids.append(len(seqs))
+
+        if self._vocab:
+            tokens = self._vocab[tokens]
+        # The valid length of sentences. Only real  tokens are attended to.
+        valid_length = len(tokens)
+        return tokens, segment_ids, valid_length, p_mask
+
+
+class BertTStyleSentenceTransform:
+    r"""BERT style data transformation.
+
+    Parameters
+    ----------
+    tokenizer : BERTTokenizer.
+        Tokenizer for the sentences.
+    max_seq_length : int.
+        Maximum sequence length of the sentences.
+    vocab : Vocab
+        The vocabulary which has cls_token and sep_token registered.
+        If vocab.cls_token is not present, vocab.bos_token is used instead.
+        If vocab.sep_token is not present, vocab.eos_token is used instead.
+    left_cls : bool
+        Insert [CLS] to the start/end of the sequence
+    """
+    def __init__(self,
+                 tokenizer,
+                 max_seq_length=None,
+                 vocab=None):
+        assert tokenizer.vocab or vocab
+        self.Truncate = TruncateTransform(max_len=max_seq_length)
+        self._tokenizer = tokenizer
+        self._vocab = tokenizer.vocab if vocab is None else vocab
```

Comment:
Instead of trying to infer the `seperators` for `ConcatSeqTransform` based on the BERT vs RoBERTa heuristic with `vocab`, let's just make it explicit and ask the user to specify the separators?